test/test.js

```diff
@@ -19083,6 +19083,29 @@
 
   /*--------------------------------------------------------------------------*/
 
+  QUnit.module('lodash.toSafeInteger');
+
+  (function() {
+    QUnit.test('should convert values to safe integers', function(assert) {
+      assert.expect(6);
+
+      assert.strictEqual(_.toSafeInteger(-5.6), -5);
+      assert.strictEqual(_.toSafeInteger('5.6'), 5);
+      assert.strictEqual(_.toSafeInteger(), 0);
+      assert.strictEqual(_.toSafeInteger(NaN), 0);
```

Comment:
If we're following that spec, why do we want to deviate on infinity and
-infinity?

On Thu, Oct 22, 2015 at 10:34 AM, Xotic750 <notifications@github.com> wrote:

> In test/test.js
> <https://github.com/lodash/lodash/pull/1555#discussion_r42755369>:
>
> > @@ -19083,6 +19083,29 @@
> >
> >    /*--------------------------------------------------------------------------*/
> >
> > +  QUnit.module('lodash.toSafeInteger');
> > +
> > +  (function() {
> > +    QUnit.test('should convert values to safe integers', function(assert) {
> > +      assert.expect(6);
> > +
> > +      assert.strictEqual(_.toSafeInteger(-5.6), -5);
> > +      assert.strictEqual(_.toSafeInteger('5.6'), 5);
> > +      assert.strictEqual(_.toSafeInteger(), 0);
> > +      assert.strictEqual(_.toSafeInteger(NaN), 0);
>
> toInteger is based on
> http://www.ecma-international.org/ecma-262/6.0/#sec-tointeger and
> toSafeInteger is based on toInteger
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lodash/lodash/pull/1555/files#r42755369>.
>